### PR TITLE
Don't create default config dir if --configuration used

### PR DESCRIPTION
**Pull request by watsonian**
This solves an issue where the CLI would always attempt to create the default config directory (`$HOME/.doppler`) even if a different location was specified using `--configuration`. Now, we provide a `--config-dir` flag that will set the entire directory being used. It also deprecates the `--configuration` flag, which was around from before we used a config directory (there is no real reason to have more than one config FILE).

### CODE REVIEW

File: pkg/configuration/config.go [hunk 1]
```diff
@@ -59,7 +59,12 @@ func init() {
 }
 
 // Setup the config directory and config file
-func Setup() {
+func Setup(cmd *cobra.Command) {
+	if cmd.Flags().Changed("config-dir") {
+		UserConfigFile = filepath.Join(UserConfigDir, configFileName)
+	}
```
Comment (Piccirello) [inline pkg/configuration/config.go]: Is this change needed if you're setting the `UserConfigDir` and `UserConfigFile` in root's `loadFlags()` function?

Comment (watsonian) [reply pkg/configuration/config.go]: I'm not familiar enough with Cobra to know if it _should_ be required, but I know in practice it seems to be. With that change backed out:

```
❯ ./doppler secrets -p example -c dev --config-dir /tmp/doppler --debug
Debug: Using config dir /tmp/doppler
Debug: Using config file /Users/watsonian/.doppler/.doppler.yaml
```

with the change in place:

```
❯ ./doppler secrets -p example -c dev --config-dir /tmp/doppler --debug
Debug: Using config dir /tmp/doppler
Debug: Using config file /tmp/doppler/.doppler.yaml
```

Comment (Piccirello) [reply pkg/configuration/config.go]: I pushed up a [change](https://github.com/DopplerHQ/cli/compare/9ba954ce6859fdcc34e64a7a0060047273a7e403..4ba5985d517e2d11dffd3561e089798b2d12d035) to simplify this a little bit. I also added some tests to verify that `--config-dir` and `--configuration` play nicely.

**Review by Piccirello — APPROVED** [review]

**Review by nmanoogian — APPROVED** [review]
